File: 2021/102.binary-tree-level-order-traverse.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
    res := [][]int{}
    dfs(root, 0, &res)
    return res
}
func dfs(node *TreeNode, depth int, res *[][]int) {
    if node == nil {
        return
    }
    if len(*res) <= depth {
        *res = append(*res, []int{})
    }
    (*res)[depth] = append((*res)[depth], node.Val)
    dfs(node.Left, depth + 1, res)
    dfs(node.Right, depth + 1, res)
}

File: 2021/python/102.binary-tree-level-order-traverse.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        
        def dfs(node, depth, res):
            if not node:
                return
            if depth >= len(res):
                res.append([])
            res[depth].append(node.val)
            dfs(node.left, depth + 1, res)
            dfs(node.right, depth + 1, res)
        
        res = []
        dfs(root, 0, res)
        return res
